### ISSUE TITLE
Spamassassin bayes database location (amavis vs. root)

### ISSUE THREAD

**Issue by makloda** [tomav/docker-mailserver#365]
The FAQ states running sa-learn user docker exec -it which runs inside the docker container using root, hence the bayes database is built in the /root/.spamassassin path inside the container.

I am not 100% sure, but I believe Amavis is calling Spamassassin using the amavis user, and therefore looking in the wrong location for the bayes db:

```
Oct 26 18:28:19.018 [9232] dbg: config: using "/var/lib/amavis/.spamassassin" for user state dir
Oct 26 18:28:19.019 [9232] dbg: bayes: tie-ing to DB file R/O /var/lib/amavis/.spamassassin/bayes_toks
Oct 26 18:28:19.026 [9232] dbg: bayes: tie-ing to DB file R/O /var/lib/amavis/.spamassassin/bayes_seen
(snipped from spamassassin -t -D output)
```

Please take this with a grain of salt as I'm a docker and unix novice, but I was thus far unable to get any bayes scoring in my received emails. All other spam assassin scoring works fine.

I will try overriding the bayes db path next and see if that works.

**Comment by tyranron** [comment]
Yeap, you're right. AMaViS is actually an implementation of SpamAssassin daemon. More details [described here](https://github.com/tomav/docker-mailserver/issues/311).

Try to specify [`--dbpath` option](https://spamassassin.apache.org/full/3.2.x/doc/sa-learn.html) for `sa-learn` command.

**Comment by makloda** [comment]
OK I can confirm using the database path option worked for me, I am now getting Bayes scoring. Maybe this could be added to the FAQ for those interested.

Thanks for this fantastic docker image!

**Comment by tyranron** [comment]
Hmm...@tomav may be we can hack this moment a bit?
```
rm -rf /root/.spamassassin
ln -s /var/lib/amavis/.spamassassin /root/.spamassassin
```

**Comment by tomav** [comment]
Are you using `ONE_DIR=1`?

**Comment by makloda** [comment]
For any amavis/spamassassin noobs who get confused coming from Mailscanner (like myself) there's lots of introductory info here:

https://wiki.apache.org/spamassassin/IntegratedInPostfixWithAmavis
https://www.ijs.si/software/amavisd/#faq-spam

**Comment by makloda** [comment]
I used ONE_DIR=1 now which works well. Also I mapped /etc/amavis/conf.d to a host folder to have control over amavis file extension (virus) filtering in a more flexible way.

Bayes is working now, own rules are working and bayes token updates are cronned using:

`/usr/local/bin/docker exec --user amavis mailserver sa-learn --spam /var/mail/domain.com/user/.Junk --dbpath /var/lib/amavis/.spamassassin`

Another suggestion for the FAQ: it didn't mention sa-learn'ing ham, however, without at least 200 learned hams Bayes will spit out an error message and will not contribute its scoring.
